src/Microsoft.HttpRepl/Commands/BaseHttpCommand.cs

```diff
@@ -92,6 +94,21 @@ public override CommandInputSpecification InputSpec
 
         protected override async Task ExecuteAsync(IShellState shellState, HttpState programState, DefaultCommandInput<ICoreParseResult> commandInput, ICoreParseResult parseResult, CancellationToken cancellationToken)
         {
+            if (programState is null)
```

Comment:
This 4-line pattern comes up a lot, how do you feel about the more concise `programState = programState ?? throw new ...`?

Reply:
I've never really been a huge fan of it, since in the common case it means (from a code read perspective, not an execution perspective, I assume) you've got an extra self-assignment that you have to ignore. But I agree those four-line blocks were repetitive and annoying. I changed it.